include/http/request_handler.js

```diff
@@ -762,33 +762,32 @@ module.exports = function RequestHandlerModule(pb) {
         this.session = session;
 
         var hostname = this.hostname;
-        var siteObj = RequestHandler.sites[hostname],
-            redirectHost = RequestHandler.redirectHosts[hostname];
-
-        //derive the localization. We do it here so that if the site isn't
-        //available we can still have one available when we error out
-        this.localizationService = this.deriveLocalization({ session: session });
+        var siteObj = RequestHandler.sites[hostname];
+        var redirectHost = RequestHandler.redirectHosts[hostname];
 
         // If we need to grab a host redirect
         if (!siteObj && redirectHost && RequestHandler.sites[redirectHost]) {
             siteObj = RequestHandler.sites[redirectHost];
         }
+        this.siteObj = siteObj;
 
-        else if (!siteObj) {
+        //derive the localization. We do it here so that if the site isn't
+        //available we can still have one available when we error out
+        this.localizationService = this.deriveLocalization({ session: session });
```

Comment:
This change isn't part of the section stuff but a fix for ensuring that a site is limited to its assigned locales after the redirect changes went in.